### PR TITLE
[ISSUE #4226] Message length exceeds the maximum length when sendback

### PR DESCRIPTION
**Pull request by ni-ze**
Make consistent semantics of maxMessageSize, refers to the message body size.

In current version, maxMessageSize refers to message total length not only include message body but also include others fields, such as magic code, queueId, body crc, properties etc.

If we make length limit base on message body, we can get consistent semantics of maxMessageSize easily.

### CODE REVIEW

File: store/src/main/java/org/apache/rocketmq/store/config/MessageStoreConfig.java [hunk 1]
```diff
@@ -87,8 +87,8 @@ public class MessageStoreConfig {
     private int fileReservedTime = 72;
     // Flow control for ConsumeQueue
     private int putMsgIndexHightWater = 600000;
-    // The maximum size of message,default is 4M
-    private int maxMessageSize = 1024 * 1024 * 4;
+    // The maximum size of message body,default is 4M,4M only for body length,not include others.
+    private int maxMessageBodySize = 1024 * 1024 * 4;
```
[the remark this answers is not in the document]
Comment (ni-ze) [reply store/src/main/java/org/apache/rocketmq/store/config/MessageStoreConfig.java:91]: Keep it as before is ok, but I do not get why it incompatibility with before version, I do not modify it in client.

[the remark this answers is not in the document]
Comment (areyouok) [reply store/src/main/java/org/apache/rocketmq/store/config/MessageStoreConfig.java:91]: If someone deploy broker and set maxMessageSize=8M, the setting has no effect after upgrade.

[the remark this answers is not in the document]
Comment (ni-ze) [reply store/src/main/java/org/apache/rocketmq/store/config/MessageStoreConfig.java:91]: Thx, Got it.

**Review by duhenglucky — APPROVED** [review]

**Review by dongeforever — COMMENTED** [review]
The code seems OK.

And it is better to add some tests,  Sending a message with the body size equal to the maxMessageSize should succeed.

**Review by areyouok — APPROVED** [review]

File: broker/src/main/java/org/apache/rocketmq/broker/processor/EndTransactionProcessor.java [hunk 2]
```diff
@@ -55,7 +55,7 @@ public RemotingCommand processRequest(ChannelHandlerContext ctx, RemotingCommand
         RemotingCommandException {
         final RemotingCommand response = RemotingCommand.createResponseCommand(null);
         final EndTransactionRequestHeader requestHeader =
-            (EndTransactionRequestHeader)request.decodeCommandCustomHeader(EndTransactionRequestHeader.class);
+            (EndTransactionRequestHeader) request.decodeCommandCustomHeader(EndTransactionRequestHeader.class);
```
[the remark this answers is not in the document]
Comment (ni-ze) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/EndTransactionProcessor.java:58]: fixed.

File: store/src/main/java/org/apache/rocketmq/store/CommitLog.java [hunk 3]
```diff
@@ -85,12 +85,12 @@ public CommitLog(final DefaultMessageStore defaultMessageStore) {
         String storePath = defaultMessageStore.getMessageStoreConfig().getStorePathCommitLog();
         if (storePath.contains(MessageStoreConfig.MULTI_PATH_SPLITTER)) {
             this.mappedFileQueue = new MultiPathMappedFileQueue(defaultMessageStore.getMessageStoreConfig(),
-                    defaultMessageStore.getMessageStoreConfig().getMappedFileSizeCommitLog(),
```
[the remark this answers is not in the document]
Comment (ni-ze) [reply store/src/main/java/org/apache/rocketmq/store/CommitLog.java]: Thanks for your review, I have fixed it and keep it as it is.

**Comment by ni-ze** [comment]
> 
sure, I will add a unit test.

**Pull request merged**